godot-core/src/obj/traits.rs

```diff
@@ -210,6 +210,22 @@ pub trait IndexEnum: EngineEnum {
 /// Trait that's implemented for user-defined classes that provide a `Base<T>` field.
 ///
 /// Gives direct access to the containing `Gd<Self>` from `Self`.
+///
+/// # Using WithBaseField as a bound
+///
+/// In order to call `self.base()` or `self.base_mut()` within a trait or on a type you define, the type of `Self::Base` must be specified via `GodotClass<Base = T>`
+///
+/// E.g.
+///
+/// ```no_run
+/// fn some_fn<T>(gd: Gd<T>)
+/// where
+///     T: WithBaseField + GodotClass<Base = Node3D>,
```

Comment:
```suggestion
///     T: WithBaseField<Base = Node3D>,
```
`WithBaseField` requires `GodotClass`, so we don't need to declare both -- we can even specify `Base` through `WithBaseField`, even though `Base` is a part of `GodotClass` and not `WithBaseField` (again, due to their relationship).